File: DocumentAnalysisSystem/Readers/DocumentReaderFactory.cs
using System;
using System.IO;

namespace DocumentAnalysisSystem.Readers
{
    public static class DocumentReaderFactory
    {
        public static IDocumentReader CreateReader(string filePath)
        {
            var extension = Path.GetExtension(filePath)?.ToLowerInvariant();
            return extension switch
            {
                ".txt" => new TxtDocumentReader(),
                ".docx" => new DocxDocumentReader(),
                ".pdf" => new PdfDocumentReader(),
                _ => throw new NotSupportedException("Định dạng file không được hỗ trợ."),
            };
        }
    }
}

File: DocumentAnalysisSystemJava/src/main/java/com/documentanalysis/readers/DocumentReaderFactory.java
package com.documentanalysis.readers;

import java.nio.file.Path;
import java.nio.file.Paths;

public class DocumentReaderFactory {
    public static IDocumentReader createReader(String filePath) {
        String extension = getExtension(filePath).toLowerCase();
        return switch (extension) {
            case ".txt" -> new TxtDocumentReader();
            case ".docx" -> new DocxDocumentReader();
            case ".pdf" -> new PdfDocumentReader();
            default -> throw new IllegalArgumentException("Unsupported file format: " + extension);
        };
    }

    private static String getExtension(String filePath) {
        Path path = Paths.get(filePath);
        String fileName = path.getFileName().toString();
        int dotIndex = fileName.lastIndexOf('.');
        if (dotIndex == -1) {
            return "";
        }
        return fileName.substring(dotIndex);
    }
}
